test/functional/test_framework/authproxy.py

```diff
@@ -131,19 +133,26 @@ def get_request(self, *args, **argsn):
 
     def __call__(self, *args, **argsn):
         postdata = json.dumps(self.get_request(*args, **argsn), default=EncodeDecimal, ensure_ascii=self.ensure_ascii)
-        response = self._request('POST', self.__url.path, postdata.encode('utf-8'))
+        response, status = self._request('POST', self.__url.path, postdata.encode('utf-8'))
         if response['error'] is not None:
-            raise JSONRPCException(response['error'])
+            raise JSONRPCException(response['error'], status)
         elif 'result' not in response:
             raise JSONRPCException({
-                'code': -343, 'message': 'missing JSON-RPC result'})
+                'code': -343, 'message': 'missing JSON-RPC result'}, status)
+        elif status != HTTPStatus.OK:
+            raise JSONRPCExceptoin({
```

Comment:
`JSONRPCExceptoin` does not exist :-)

Reply:
So we are testing that this unexpected situation does not occur by means of Python throwing an error otherwise. :)

Good catch, fixed!